execution/limits/limits.go

```diff
@@ -0,0 +1,68 @@
+package limits
+
+import (
+	"sync"
+
+	"github.com/efficientgo/core/errors"
+	"github.com/thanos-io/promql-engine/query"
+)
+
+type Limits struct {
```

Comment:
I think this should be an interface and we should have a `nop` implementation to avoid going through locks when limits are not configured.

Reply:
Removed the locks but there is a short circuit if we have no limit at the start anyhow!